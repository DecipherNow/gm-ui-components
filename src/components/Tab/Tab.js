import React from "react";
import { PropTypes } from "prop-types";

import TabWrap from "./components/TabWrap";
import TabLabel from "./components/TabLabel";

const Tab = ({
  label = "",
  active = false,
  disabled = false,
  children,
  clickAction
}) => {
  return (
    <TabWrap active={active} disabled={disabled} onClick={clickAction}>
      {children}
      <TabLabel>{label}</TabLabel>
    </TabWrap>
  );
};

Tab.propTypes = {
  active: PropTypes.bool,
  children: PropTypes.any,
  clickAction: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  label: PropTypes.string
};

export default Tab;
