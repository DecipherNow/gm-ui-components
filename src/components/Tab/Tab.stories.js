import React from "react";
import { storiesOf } from "@storybook/react";
import { text, boolean } from "@storybook/addon-knobs";

import Tab from "./Tab";

const stories = storiesOf("Components|Tabs", module);

stories
  .add(
    "Default",
    () => {
      return (
        <Tab
          clickAction={() => alert("hello")}
          label={text("label", "label")}
          active={boolean("active", true)}
          disabled={boolean("disabled")}
        />
      );
    },
    {
      info: {
        text:
          "An individual Tab component that is meant to be used as a child of TabGroup."
      }
    }
  )
  .add(
    "Typical",
    () => {
      return (
        <Tab
          clickAction={() => alert("hello")}
          label={text("label", "Filters")}
          active={boolean("active", false)}
          disabled={boolean("disabled", false)}
        />
      );
    },
    {
      info: {
        text:
          "An individual Tab component that is meant to be used as a child of TabGroup."
      }
    }
  );
