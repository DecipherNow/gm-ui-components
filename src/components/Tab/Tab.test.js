import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import Tab from "./Tab";

describe("Tab", () => {
  it("matches snapshot", () => {
    const aTab = renderer.create(<Tab label="Test" />).toJSON();
    expect(aTab).toMatchSnapshot();
  });
});
