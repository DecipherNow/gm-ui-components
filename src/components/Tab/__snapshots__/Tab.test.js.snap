// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab matches snapshot 1`] = `
.c0 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding-left: 8px;
  padding-right: 8px;
  font-weight: 600;
  border-bottom: 2px solid;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0:hover {
  border-color: currentColor;
}

.c0:active {
  border-color: #0aab2a;
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<a
  className="c0"
  disabled={false}
  onClick={undefined}
>
  <span
    className="c1"
  >
    Test
  </span>
</a>
`;
